### ISSUE TITLE
nest generate command not working on typescript@5.5.4

### ISSUE THREAD

**Issue by AlangGY** [nestjs/nest-cli#2662]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

1. I made a new nest project using nestcli 
  - command: `nest new {project-name}`
<img width="903" alt="image" src="https://github.com/user-attachments/assets/48931b48-c18b-4076-9b67-e9325904b4c7">

3. I tried to create a module named `test`, but It fails.
  - command: `nest g module {test}`
<img width="923" alt="image" src="https://github.com/user-attachments/assets/e2e713cf-bfc3-4efe-b8a6-acca4be4e6c9">

### Minimum reproduction code

https://github.com/AlangGY/nest-cli-error-report

### Steps to reproduce

1. pnpm i
2. npx nest generate module test

### Expected behavior

I expect to generate `test` module using `nest g` command as usual.

### Package version

10.4.2

### NestJS version

10.3.10

### Node.js version

21.0.0

### In which operating systems have you tested?

- [X] macOS
- [ ] Windows
- [ ] Linux

## How To Solve

I found that the problem was with TypeScript@5.5.4, which was released on 2024/07/23 (or 22). After downgrading to TypeScript@5.5.3, the nest g command worked as expected.

| Problem | Solved |
| ---- | ---- |
| <img width="910" alt="image" src="https://github.com/user-attachments/assets/f7bab151-0188-4cd1-82fe-07acdcd158ae"> | <img width="908" alt="image" src="https://github.com/user-attachments/assets/a0de3f07-dbaa-41f9-9038-a8bf0de5e04b"> | 
| <img width="911" alt="image" src="https://github.com/user-attachments/assets/14545a7d-912e-44f8-80e1-a9a0e030f938"> | <img width="911" alt="image" src="https://github.com/user-attachments/assets/94e36360-077e-49e2-98b7-55ad985a3797"> |

I hope the version of @nestjs/cli that also corresponds to typescript@5.5.4 is updated.

**Comment by yuhengAlioth** [comment]
I updated the latest cli version locally for windows, but the previous project did not use commands, and the cli version in my newly created project is 10.0.0

**Comment by yuhengAlioth** [comment]
The same command cannot be used

**Comment by summer-boythink** [comment]
me too

**Comment by zakhaev26** [comment]
same issue while generating any module.

**Comment by 112frosta** [comment]
known any fix?

**Comment by GDR** [comment]
> known any fix?

Just set in your package.json deps:
```
"deps": {
  ...
  "typescript": "=5.5.3",
  ...
}
```

**Comment by micalevisk** [comment]
@kamilmysliwiec this seems to be related with this `node.getChildren()` invokation: https://github.com/nestjs/schematics/blob/0e6458dc7519060ef2ee52b7083132eb5d8443d1/src/utils/metadata.manager.ts#L124C33-L124C44

not sure how to address that tho

**Comment by yuhengAlioth** [comment]
> > known any fix?   知道什么修复方法吗?
> 
> Just set in your package.json deps:把你的包裹放进去。json deps:
> 
> ```
> "devDependencies": {
>   ...
>   "typescript": "=5.5.3",
>   ...
> }
> ```

of no use

**Comment by marcasteroid** [comment]
Got  same problem here.
`nest g res orders --no-spec`

```
Error: Cannot read properties of undefined (reading 'text')

Failed to execute command: node @nestjs/schematics:resource --name=orders --no-dry-run --no-skip-import --language="ts" --source-root="src" --no-spec --no-flat --spec-file-s
```uffix="spec"

with `"typescript": "^5.1.3"`.

**Comment by marcasteroid** [comment]
> https://github.com/nestjs/schematics/blob/0e6458dc7519060ef2ee52b7083132eb5d8443d1/src/utils/metadata.manager.ts#L124C33-L124C44

It works. Thank a ton.

**Comment by zqp12s** [comment]
Thank you very much

**Comment by HuskkyQ** [comment]
Bro thank you so much.

**Comment by alexlionnel** [comment]
I have the same issue:
Node: v20.13.1
Nest: 10.4.2
typescripot: 5.5.3

I tried to execute this command
```nest generate module user```

but i got the same error
```
Error: Cannot read properties of undefined (reading 'text')

Failed to execute command: node @nestjs/schematics:module --name=user --no-dry-run --no-skip-import --language="ts" --source-root="src" --spec --no-flat --spec-file-suffix="spec"
```

This is my ``package.json`` file

```
{
  "name": "nestjs-chat",
  "version": "0.0.1",
  "description": "",
  "author": "",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "build": "nest build",
    "format": "prettier --write \"src/**/*.ts\" \"test/**/*.ts\"",
    "start": "nest start",
    "start:dev": "nest start --watch",
    "start:debug": "nest start --debug --watch",
    "start:prod": "node dist/main",
    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
    "test": "jest",
    "test:watch": "jest --watch",
    "test:cov": "jest --coverage",
    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
    "test:e2e": "jest --config ./test/jest-e2e.json"
  },
  "dependencies": {
    "@nestjs/common": "^10.3.10",
    "@nestjs/core": "^10.3.10",
    "@nestjs/platform-express": "^10.3.10",
    "npm-check-updates": "^16.14.20",
    "reflect-metadata": "^0.2.0",
    "rxjs": "^7.8.1"
  },
  "devDependencies": {
    "@nestjs/cli": "^10.4.2",
    "@nestjs/schematics": "^10.1.2",
    "@nestjs/testing": "^10.3.10",
    "@types/express": "^4.17.17",
    "@types/jest": "^29.5.2",
    "@types/node": "^20.3.1",
    "@types/supertest": "^6.0.0",
    "@typescript-eslint/eslint-plugin": "^7.0.0",
    "@typescript-eslint/parser": "^7.0.0",
    "eslint": "^8.42.0",
    "eslint-config-prettier": "^9.0.0",
    "eslint-plugin-prettier": "^5.0.0",
    "jest": "^29.5.0",
    "prettier": "^3.0.0",
    "source-map-support": "^0.5.21",
    "supertest": "^7.0.0",
    "ts-jest": "^29.1.0",
    "ts-loader": "^9.4.3",
    "ts-node": "^10.9.1",
    "tsconfig-paths": "^4.2.0",
    "typescript": "^5.5.3"
  },
  "jest": {
    "moduleFileExtensions": [
      "js",
      "json",
      "ts"
    ],
    "rootDir": "src",
    "testRegex": ".*\\.spec\\.ts$",
    "transform": {
      "^.+\\.(t|j)s$": "ts-jest"
    },
    "collectCoverageFrom": [
      "**/*.(t|j)s"
    ],
    "coverageDirectory": "../coverage",
    "testEnvironment": "node"
  }
}
```

**Comment by kamilmysliwiec** [comment]
Fixed in `@nestjs/schematics@10.1.3`
